xds/internal/resolver/serviceconfig.go

```diff
@@ -142,6 +143,8 @@ type configSelector struct {
 }
 
 var errNoMatchedRouteFound = status.Errorf(codes.Unavailable, "no matched route was found")
+var errRouteActionUnsupported = status.Errorf(codes.Unavailable, "received route was of type RouteActionUnsupported, require type RouteActionRoute")
```

Comment:
These types are not the types that are configured in xDS, and aren't right to be using in the error messages here.

If we want to error with the proper type, we need to either define a `xdsresource.RouteActionType` for every xds action type, or we need to plumb in the actual xds route action enum.

Reply:
hi @dfawley 
do you mean that instead of a generic `RouteActionUnsupported` we need to return a more precise `RouteActionType`? 
if so could you guide me on how to identify the other possible xds route actions?

cc @easwars

Reply:
@easwars is out of the office for a couple weeks. Let's just go ahead and change it.  The only way to improve it beyond that would be to pass the actual xds enum type down to this layer, but that doesn't seem worth doing just to have a slightly better error message.

Reply:
sure @dfawley, I'll implement it with the router check changes

@zasweq could you please give the confirmation for the newInterceptor() changes here - https://github.com/grpc/grpc-go/pull/6248#issuecomment-1582237029